src/cpp/session/SessionMain.cpp

```diff
@@ -1654,6 +1655,9 @@ Error rInit(const rstudio::r::session::RInitInfo& rInitInfo)
    s_waitForMethodNames.push_back(kChooseFileCompleted);
    s_waitForMethodNames.push_back(kUserPromptCompleted);
    s_waitForMethodNames.push_back(kHandleUnsavedChangesCompleted);
+   
+   // initialize lockfile management
+   FileLock::initialize();
```

Comment:
I think this belongs much earlier on in `main` since it's a process-wide service we are initializing. It should go right before we initialize the HttpConnectionListener here: https://github.com/rstudio/rstudio/blob/feature/directory-file-locking/src/cpp/session/SessionMain.cpp#L3151